packages/@react-spectrum/layout/src/flex.css

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+.flex-container {
+  /* this is necessary so that the inner margins don't affect anything outside */
+  overflow: hidden;
```

Comment:
This is going to show that problem with button focus rings pretty easily

Reply:
yeah, the problem is the inner div has a negative margin which affects the height of the outer div. not sure if there is another way to do this.